src/cmps/AppStateManager.tsx

```diff
@@ -1,18 +1,53 @@
-import { useState } from "react";
-import Header from "./Header";
-import PokemonContent from "./PokemonContent";
+import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
 import GlobalStyle from "../assets/globalStyles/GlobalStyle";
+import Header from "../cmps/Header";
+import PokemonContent from "../cmps/PokemonContent";
+import FightArenaPage from "../pages/FightArenaPage";
+import { useEffect, useState } from "react";
+import { Pokemon } from "../interfaces/Pokemon";
 
-const AppStateManager = () => {
+function AppStateManager() {
   const [selectedCtg, setSelectedCtg] = useState(0);
+  const [pokemonData, setPokemonData] = useState<Pokemon[]>([]);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const fetchData = async () => {
+    const response = await fetch("/src/data/pokemon.json");
+    const data = await response.json();
+    setPokemonData(data);
+  };
+
+  const userPokemons = pokemonData.filter((pokemon) => pokemon.belongsToUser);
 
   return (
-    <>
+    <Router>
       <GlobalStyle />
       <Header setSelectedCtg={setSelectedCtg} selectedCtg={selectedCtg} />
-      <PokemonContent selectedCtg={selectedCtg} />
-    </>
+      <Routes>
+        <Route
+          path="/"
+          element={
+            <PokemonContent
+              selectedCtg={selectedCtg}
+              pokemonData={pokemonData}
```

Comment:
also makes more sense this api calls will happen inside the component and not from the router component